app/lib/three_scale/analytics/user_tracking.rb

```diff
@@ -26,16 +27,26 @@ def respond_to_missing?(method_sym, include_all)
           end
 
           def method_missing(method_sym, *args, &block)
-            DELEGATED_METHODS.include?(method_sym) || super
+            return super unless DELEGATED_METHODS.include?(method_sym)
+
+            Rails.logger.info("#{method_sym} call avoided. Segment disabled")
           end
         end
 
-        def segment_configured?
-          defined?(::Segment)
+        def self.segment_configured?
+          System::Application.config.three_scale.segment.enabled && defined?(::Segment)
```

Comment:
I think this should be from the same config that it has in the instance, like this:

```ruby
def segment_configured?
  config.enabled && defined?(::Segment)
end
```

![image](https://user-images.githubusercontent.com/11318903/59856292-ea6dd000-9376-11e9-9ca6-7f6554d8d188.png)

![image](https://user-images.githubusercontent.com/11318903/59856320-f5c0fb80-9376-11e9-9fe0-542f09123c5a.png)

Reply:
💯